tests/test_helpers.py

```diff
@@ -0,0 +1,59 @@
+import pytest
+
+from globus_automate_client import helpers
+
+
+@pytest.mark.parametrize(
+    "args, expected, message",
+    (
+        # List
+        (([],), set(), "empty list identity not maintained"),
+        (([1],), {1}, "list identity not maintained"),
+        (([1, 1],), {1}, "in-list deduplication not performed"),
+        (([1], [1]), {1}, "cross-list deduplication not performed"),
+        (([1, 2],), {1, 2}, "unique items missing"),
+        # Dictionary
+        (({}, "k"), None, "dict with no matching key must return None"),
+        (({"k": [1]},), None, "dict w/o matching key specified must return None"),
+        (({"k": [1]}, "k"), {1}, "dict w/o list and w/ matching key must be added"),
+        (([1], {"k": [2]}, "k"), {1, 2}, "dict with matching key must be added"),
+        (([1], {"k": [1]}, "k"), {1}, "dict values must be de-duplicated"),
+        (([1], {"k": 2}, "k"), {1}, "scalar value in dict must be ignored"),
+        # Multiple values
+        (
+            ([1, 2], [2, 3], {"k1": [4], "k2": [5], "k3": [6]}, "k2", "k3", "bogus"),
+            {1, 2, 3, 5, 6},
+            "all values must be found",
+        ),
+    ),
+)
+def test_merge_lists(args, expected, message):
+    """Validate globus_automate_client.helpers.merge_lists()."""
```

Comment:
This was what I was missing as I looked at `sc-11808`.  Did you happen to run into any cases that were problematic? I still haven't had a chance to investigate/patch that bug but if that function made sense to you I would appreciate an extra set of eyes on that.

Reply:
Sure thing! I'll dig into that. I'll mark this as resolved and look into `sc-11808`.